alphapulldown/utils/distogram_parser.py

```diff
@@ -0,0 +1,99 @@
+__author__ = "Grzegorz Chojnowski"
+__email__ = "gchojnowski@embl-hamburg.de"
+__date__ = "16 Apr 2024"
+__info__ = "extracts contact information from AF2 distograms"
+
+
+import pickle, glob
+import os
+import numpy as np
+import string
+
+class distogram_parser:
+    """
+        extracts predicted contacts from AF2 distograms assuming that 
+        'seq' list in output pickle maps onto the distogram 
+            (contains all target sequences in output model/distogram order)
+
+        output: list of residues close in space [(res_A_idx, res_A_chain), (res_B_idx, res_B_chain), pbty( dist(A,B)<distance ) > pbtycutoff ]
+    """
+
+    def __init__(self):
+        pass
+
+    def get_contacts(self, directory, distance=8, pbtycutoff=0.8, cross_only=True, verbose=False):
+        """
+            selects from datadir a pkl/distogram corresponding to a top-ranked model 
+        """
+
+        model_ranking = []
+        for fn in glob.glob(os.path.join(datadir, "*.pkl")):
+            with open(fn, 'rb') as ifile:
+                d=pickle.load(ifile)
+            model_ranking.append([fn, d.get('ranking_confidence',0)])
+
+        top_ranked_fn = sorted(model_ranking, key=lambda x:x[1])[-1]
+
+        if verbose: 
+            print(f"Selected {os.path.basename(top_ranked_fn[0])} with ranking confidence {top_ranked_fn[1]:.2f}")
+
+        # reparse top ditogram; avoids storing all pickles in memory         
+        with open(fn, 'rb') as ifile:
+            d=pickle.load(ifile)   
```

Comment:
do not reat pkls second time (cache them from the previous loop) - sometimes they are big and if you have many, it can be quite slow